In [1]:
# import tqdm
# from torch.utils.data import Dataset
# from collections import defaultdict
# import random

# class MSMARCODataset(Dataset):
#     def __init__(self, filepath):
#         self.qid_qtxt = {}
#         self.qid_did = defaultdict(list)
#         self.did_dtxt = {}

#         with open(filepath, 'r', encoding='utf-8') as file:
#             for i, line in enumerate(tqdm.tqdm(file)):
#                 q_id, doc_id, query, doc = line.split('\t')
#                 self.qid_qtxt[q_id] = query
#                 self.qid_did[q_id].append(doc_id)
#                 self.did_dtxt[doc_id] = doc
#         print(i+1)
    # def __len__(self):
    #     return len(self.data)

    # def __getitem__(self, idx):
    #     return self.data[idx]
    
    # def get_data_for_query_id(self, query_id):
    #     return [item for item in self.data if item['query_id'] == query_id]
    
    # def get_query_id_tuples(self):
    #     return [(item['query_id'], item['query']) for item in self.data]

# our_data = MSMARCODataset('top1000.dev')

6668967it [00:11, 600067.52it/s]

6668967


In [2]:
# '188714'
# '408149'

In [3]:
# len(our_data.qid_qtxt)

6980

In [4]:
# len(our_data.did_dtxt)

3895239

There are 6980 unique queries, 3895239 unique documents, and 6668967 document-query pairs in total.

In [5]:
# len(our_data.qid_did['188714'])

1000

Each query is linked to ~1000 documents

In [7]:
rel_qid_did = defaultdict(list)
# only filled with docs that are relevant to a query
with open('qrels.dev.tsv', 'r') as file:
    for i,line in enumerate(file):
        a,b,c,d = line.strip().split('\t')
        rel_qid_did[a].append(c)
    print(i+1)

59273


In [8]:
# len(rel_qid_did)

55578

There are 55578 unique queries. Each query in the qrel dataset has ~1 relevant document

In [9]:
# # 55 578
# qids_rel = list(rel_qid_did.keys())
# # 6980
# qids_our = list(our_data.qid_did.keys())

In [10]:
# len(set(qids_rel) - set(qids_our))

48598

Each query in our dataset is in the qrels dataset!

In [11]:
# print(rel_qid_did['188714'])
# print(our_data.qid_did['188714'])

['8003843', '4321745', '8003849']
['1000052', '1022490', '1051362', '1090909', '1097195', '1130809', '1148823', '1160714', '1200661', '1241103', '1500656', '1656996', '1687849', '1751526', '1831803', '1899272', '1996465', '2133570', '2141476', '2165254', '2287600', '2455992', '2456320', '2470190', '2517166', '2666492', '2788357', '3045191', '3045313', '3045314', '3054173', '3125534', '3125543', '3172505', '3174245', '3181392', '3241451', '3483672', '3548230', '3615882', '3719372', '3849449', '386069', '3928996', '4057562', '4057564', '4074394', '4283881', '4321739', '4321742', '4405570', '443989', '454883', '454884', '4585316', '46209', '471210', '4963488', '4963491', '4963495', '496432', '5007414', '5072224', '507420', '5092333', '523571', '5499899', '5499901', '5548810', '5614247', '5624275', '5631262', '5681536', '5703711', '5734858', '5804838', '5820625', '6039084', '6043639', '6167786', '6172028', '6176398', '6192043', '6254756', '6301922', '6416264', '6420874', '6507763', '655551

In [12]:
# random_qid_ours = random.choice(list(our_data.qid_did.keys()))

# for d in rel_qid_did[random_qid_ours]:
#     if d not in our_data.qid_did[random_qid_ours]:
#         print('a relevant doc for this query is not in our dataset')

In [14]:
# MAAK EEN KOPIE VAN DE DATAFILE EN NOEM HEM ZO (HIER KOMT DE BINAIRE LABEL INFO BIJ): 
filepath = 'top1000_copy.dev'

# Read the existing content of the file
with open(filepath, 'r', encoding='utf-8') as file:
    lines = file.readlines()

# Modify each line by adding "0" at the end
modified_lines = []
for line in lines:
    q_id, doc_id, query, doc = line.split('\t')
    linez = line.strip()  # Remove any trailing newline characters
    if doc_id in rel_qid_did[q_id]:
        new_line = f"{linez}\t1\n"  # Append '1' separated by a tab and add a newline
    else:
        new_line = f"{linez}\t0\n"  # Append '0' separated by a tab and add a newline
    modified_lines.append(new_line)

# Write the modified lines back to the file
with open(filepath, 'w', encoding='utf-8') as file:
    file.writelines(modified_lines)

In [3]:
# zerocount = 0
# onecount = 0

# with open('top1000_copy.dev', 'r', encoding='utf-8') as file:
#     for line in file:
#         a,b,c,d,e = line.strip().split('\t')
#         if e == '0':
#             zerocount += 1
#         else:
#             onecount += 1

# print(zerocount)
# print(onecount)

6662962
6005
